src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -2027,10 +2025,19 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Threading\WaitSubsystem.WaitableObject.Unix.cs" />
   </ItemGroup>
   <ItemGroup Condition="'$(FeatureCoreCLR)' != 'true' and '$(TargetsWindows)' == 'true'">
-    <Compile Include="$(MSBuildThisFileDirectory)System\Threading\Thread.Windows.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Threading\WaitHandle.Windows.cs" />
     <Compile Include="$(CommonPath)\Interop\Windows\Kernel32\Interop.Threading.cs">
       <Link>Interop\Windows\Kernel32\Interop.Threading.cs</Link>
     </Compile>
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetsUnix)' == 'true' or '$(TargetsBrowser)' == 'true'">
```

Comment:
I am not sure whether it is an improvement to move the ifdefs from here to the .cs file, for one of the low-level threading related files. (The actual problem is that the threading in CoreCLR is waiting to be refactored into managed. Once that happens, the `FeatureCoreCLR` condition will disapper around all threading related files here.)

Reply:
It is an improvement only because I need a place to keep a method from being complained about due to linker nonsense. I've been struggling for the past 2 days almost entirely trying to get the linker to not strip the function or stop complaining that it isn't there. It was suggested that I get it to work so I'm not blocked and move on. The biggest issue right now from my understanding is https://github.com/dotnet/runtime/issues/51991.

Reply:
#51991 is CoreCLR specific problem, so it would be nice to keep the ifdefs and dummy methods required to workaround it under src/coreclr.

Reply:
I can do that, but I would then like to rename these files then because they are not `Thread.Windows.cs` - they are `Thread.Mono.Windows.cs`. The issue here is the various levels and slices of the build. Some are in the project files some are in the CS files and then we have a mix of naming conventions. My expectation here is any file in the shared directory should be consumable by all runtimes. If we need to pare the file down then we should do that but not at the file level.

Reply:
Most of the FeatureCoreCLR conditions in System.Private.CoreLib.Shared.projitems, including this one, are workarounds for https://github.com/dotnet/runtime/issues/31721 . 

We have been naming the files with assumption that this issue is fixed and CoreCLR is switched to use the shared implementation. Otherwise, the naming gets really hard. For example, some of the files would have to be WindowsOrWindowsEmulatorOnUnix to capture how they are used currently, etc.

Reply:
Okay. I'll try something else.